src/pages/spawn/spawnHost/spawnHostModal/PublicKeyForm.tsx

```diff
@@ -48,6 +48,8 @@ export const PublicKeyForm: React.FC<PublicKeyFormProps> = ({
           style={{ width: 200 }}
           placeholder="Select existing key"
           onChange={selectPublicKey}
+          value={keyName}
```

Comment:
we should also add `disabled={shouldAddNewKey}` to prevent the user from selecting an existing key while adding a new. I attached a video of the edgecase adding this prop can prevent.
![new name equals old name](https://user-images.githubusercontent.com/10734386/95347401-d65f2200-088a-11eb-9743-21b1d221291e.gif)